tests/deploy/ui/test_ui.py

```diff
@@ -0,0 +1,174 @@
+import socket
+import subprocess
+import sys
+import time
+
+import httpx
+import panel as pn
+import pytest
+from playwright.sync_api import Page, expect
+
+from ragna._utils import timeout_after
+from ragna.deploy import Config
+
+from ..utils import TestAssistant
+
+
+def get_available_port():
+    with socket.socket() as s:
+        s.bind(("", 0))
+        return s.getsockname()[1]
+
+
+@pytest.fixture(scope="session")
+def api_port():
+    return get_available_port()
+
+
+@pytest.fixture(scope="session")
+def ui_port():
+    return get_available_port()
+
+
+@pytest.fixture(scope="session")
+def headed_mode(pytestconfig):
+    return pytestconfig.getoption("headed") or False
+
+
+@pytest.fixture
+def config(
+    tmp_local_root,
+):
+    config = Config(
+        local_root=tmp_local_root,
+        assistants=[TestAssistant],
+        ui=dict(port=get_available_port()),
+        api=dict(port=get_available_port()),
+    )
+    path = tmp_local_root / "ragna.toml"
+    config.to_file(path)
+    return config
+
+
+class Server:
+    def __init__(self, config):
+        self.config = config
+        self.base_url = f"http://{config.ui.hostname}:{config.ui.port}"
+
+    def server_up(self):
+        try:
+            return httpx.get(self.base_url).is_success
+        except httpx.ConnectError:
+            return False
+
+    @timeout_after(60)
+    def start(self):
+        self.proc = subprocess.Popen(
+            [
+                sys.executable,
+                "-m",
+                "ragna",
+                "ui",
+                "--config",
+                self.config.local_root / "ragna.toml",
+                "--start-api",
+                "--ignore-unavailable-components",
+                "--no-open-browser",
+            ],
+            stdout=sys.stdout,
+            stderr=sys.stderr,
+        )
+
+        while not self.server_up():
+            time.sleep(1)
+
+    def stop(self):
+        self.proc.kill()
+        pn.state.kill_all_servers()
+
+
+@pytest.fixture
+def server(config):
+    server = Server(config)
+    try:
+        server.start()
+        yield server
+    finally:
+        server.stop()
+
+
+def test_health(server, page: Page) -> None:
+    health_url = server.base_url + "/health"
+    response = page.goto(health_url)
+    assert response.ok
+
+
+def test_index(server, config, page: Page) -> None:
```

Comment:
Make the test name more expressive.

```suggestion
def test_start_chat(server, config, page: Page) -> None:
```